DESIGN_DOCS.md

```diff
@@ -0,0 +1,65 @@
+
+* **Context and Scope**: A short description of the landscape in which the new system is being built, what is actually being built. It may also say what is _not_ being built, and any assumptions.
+* **Problem Statement**: The "why." A short summary of the issue(s) that this design solves.
+    This doesn't have to be a technical problem. 
+    "Developer unhappiness" or "improved user experience" are also problems.
+* **Goals and Non-goals**:  what this solution is trying to do, and is also _not_ trying to do, in concrete terms.  Measurable goals are best.
+* **Proposal**: A high level overview, followed by a detailed description. 
+    This is where specific technology, such as language, frameworks, encryption algorithms, type of authentication, and APIs can be described.  
+    It can include diagrams such as [a system context diagram](https://en.wikipedia.org/wiki/System_context_diagram), or a [sequence diagram](https://www.geeksforgeeks.org/unified-modeling-language-uml-sequence-diagrams/).
+* **Benefits and Risks**: the reasons why this solution was chosen, and the risks this solution poses.
+    For example, the solution may be very simple, but there could performance bottlenecks above a certain threshold.
+    Another: the solution is well known and widely used, but it's not a perfect fit and requires complicated changes in one area.   
+* **Alternatives and Rationale**: discuss alternatives that were considered, and why they were rejected. 
+    Note when there are absolute requirements that the solution does not and can't meet.
+    One example might be, it's a proprietary solution but we need something open source.
+* **Sources**: sources of information that led to this design.
+* **Glossary** (optional): if you think inline links to concepts are too distracting, include a glossary section. 
+    This can be links, text or both.
+
+## Process
+The process for adding a new DSNP Design Document is much like any other Pull Request in a code base.
+1. Create a branch in this repo.
+2. Commit your design doc in your branch, based on the guidelines here.
+3. Post the design document as a Draft PR for discussion.
+4. When the PR is accepted, it can be merged.
+
+## Writing style
+Remember that the first goal of the Design Document is to explain a solution to others so they can understand it well enough to start writing stories and then implement the solution.
+The next goal is explain the solution to the public.
+Also keep in mind that these documents will be written in English, and some readers may not be fluent in English. 
+
+### General
+Be clear, be concise.
+Prefer shorter words and simpler sentences.
+Break up writing into short paragraphs with one concept each.
+Use white space, headings and different text styles to separate and highlight important topics and ideas.
+
+This document is meant to be an example of the preferred writing style.
+
+### Markdown
+* Please put each sentence on its own line in the Markdown.
```

Comment:
1. Might want to say explicitly that Markdown is the format to use -- it's implied, but not actually stated here, that that's a requirement.

2. That each-sentence-on-its-own-line thing makes the source text harder to read.  I get why it's useful when viewing diffs.  Are we sure it's worth it?  An alternative strategy would be: stick to ~76 character lines as per usual, and then when editing an *existing* design doc make an effort to minimize changes line-wise -- i.e., think about the diff you're creating, don't refill paragraphs willy-nilly, etc.  That way we don't pay the one-sentence-per-line price when the original document is written, we just pay a smaller ongoing price as we edit an existing document (and note that we'd pay a similar ongoing price either way, because if the one-sentence-per-line rule were in effect, then editors of existing documents would still have to pay attention to that rule anyway).

Reply:
Doing it by line length can makes it harder to edit too.  We used this convention for the DSNP spec and it really is useful in PR reviews.  People can use soft wrap for legibility when editing.

Reply:
Okay.  If you found it useful, then let's keep it.  People can configure their editing tools appropriately.